Ambient Lighting Module protos

charminULTRA opened the pull request:
<!-- Describe what you are intending to change -->

Creation of Ambient Lighting Module protobufs in order to control light levels and RGB of the RAK 14001 (and potentially other software controlled LEDs later).

I don't have the related code in the firmware to interact with the protobufs yet, because I don't know how to do that yet :)

## Checklist before merging

- [Y] All top level messages commented
- [N/A] All enum members have unique descriptions

charminULTRA commented:
What is a buf authentication token?

thebentern commented on meshtastic/module_config.proto:
```diff
@@ -583,6 +583,26 @@ message ModuleConfig {
   }
+
+  /*Ambient Lighting Module - Settings for control of onboard LEDs to allow users to adjust the brightness levels and respective color levels.
+Initially created for the RAK14001 RGB LED module.
+*/
+
+message AmblientLightingConfig {
+
+/*Sets the overall current for the LED, range for the RAK14001 is 1-31*/
```
Looks like the indents for this message and comments are off

charminULTRA replied:
Fixed

thebentern commented:
@charminULTRA you'll also need to add this as a oneof payload_variant option to ModuleConfig

andrekir commented:
do we need a new module instead of a getter/setter via admin like canned message and ringtone?

also if multiple lights are a possibility a light id/address field could be added.

charminULTRA commented:
> 

Is there any advantage to putting it into the Admin proto file? From an organizational standpoint, I didn't want to put it someplace that would get buried.

andrekir commented:
making a few assumptions, but looks like this belongs in external notifications and is not being added there for size constraints?

for those cases we use admin messages like we set ringtone and canned messages. there is also a multi field [example](https://github.com/meshtastic/protobufs/blob/e4396fd499769f24c265985ae0ee7be05c18f65a/meshtastic/admin.proto#L202).

advantage would be not splitting external notification into 2 modules (unless of course it does require its own). but looks to me simple enough for a more simple approach. would also be fairly easy to refactor into its own module later if needed.

charminULTRA commented:
> making a few assumptions, but looks like this belongs in external notifications and is not being added there for size constraints?
> 
> for those cases we use admin messages like we set ringtone and canned messages. there is also a multi field [example](https://github.com/meshtastic/protobufs/blob/e4396fd499769f24c265985ae0ee7be05c18f65a/meshtastic/admin.proto#L202).
> 
> advantage would be not splitting external notification into 2 modules (unless of course it does require its own). but looks to me simple enough for a more simple approach. would also be fairly easy to refactor into its own module later if needed.

Ah I see what you mean. This functionality is actually intended for use outside of the ExternalNotificationModule. I wanted the LED to turn on during boot to 50% and white (which is now in 2.1.17), and after I added that I thought it would be nice to allow users to adjust the levels and RGB settings after boot to whatever they wanted, which is why I've added these protobufs for that. 

Since it's intended to be separate functionality from the ExternalNotificationModule, and since the config settings didn't seem to make sense to place inside of Device or User or Display, my proposal is a new module that can eventually control other types of LED hardware as well.

andrekir commented:
sounds good. typically smart lights have these three settings in common, so I'd suggest:

1) on/off (enabled);
2) brightness percentage (instead of current);
3) RGB color.

although brightness could be used for on/off, having it separate allows keeping a "preferred brightness" value.

charminULTRA commented:
> sounds good. typically smart lights have these three settings in common, so I'd suggest:
> 
>     1. on/off (enabled);
> 
>     2. brightness percentage (instead of current);
> 
>     3. RGB color.
> 
> 
> although brightness could be used for on/off, having it separate allows keeping a "preferred brightness" value.

Thank you! Good suggestions. A few responses:

1 - Will see if I can implement this.
2 - My understanding (from ChatGPT...) is that the app author would implement the percentage input. Should I make a comment mentioning this, or is my understanding incorrect?
3 - Similar assumption as #2

charminULTRA commented on meshtastic/module_config.proto:
```diff
@@ -583,6 +583,26 @@ message ModuleConfig {
+Initially created for the RAK14001 RGB LED module.
+*/
+
+message AmblientLightingConfig {
```
Fixed

charminULTRA commented:
Made edits including shifting my code section to be above the payload_variant section. Also clarified in comments that app authors should use percentages.

andrekir commented:
there is no value conversion done on clients, consider users exposed to fields as they are. lots of module logic like determining current vs. brightness might not even be possible to do client side.

charminULTRA commented:
@thebentern Is there something I need to click or update now? I'm also not sure how to clear this check issue.

charminULTRA reviewed (commented):
Made the changes
